kidkode/src/main/webapp/script.js

```diff
@@ -131,10 +127,10 @@ const previousButton = document.getElementById("previous");
 const nextButton = document.getElementById("next");
 
 // Variables
-let jsonFile = "art.json"; // HARDCODED: to be changed depending on the user's favorite activity
 let currentStep = "step0"; // the step we are currently on
 var userPath = ["step0"]; // this array keeps track of the user's pathway
-let userChoice = "no"; // HARDCODED: the user's choice for the current question
+var stepQuestionNumbers = new Map(); // map that keeps track of the steps and corresponsing question number
```

Comment:
Just curious, how is stepQuestionNumbers different than userPath? Not sure if its "better" to do so, but could we just use the userPath array to count what number question the user is on? I may very well be interpreting the code wrong, but just curious if that could work！

Reply:
Ah, so I made `stepQuestionNumbers` a separate data structure because I realized that in `buildQuiz()`, each radio choice id attribute was named after the question number, and `displayResults()` uses the question number to iterate through the slide elements and calculate the results. However, I fully agree with you that using one data structure to keep track of the question number, user path and career points would make the code much cleaner! Soon I'll look into the code again so I can make the edits. :)

Reply:
Okay, no worries! I was just wondering. Since it works, lets keep it and optimize next week!

Reply:
Awesome, sounds like a plan!